src/mono/wasm/runtime/gc-handles.ts

```diff
@@ -93,7 +55,9 @@ export function mono_wasm_get_js_handle(js_obj: any): JSHandle {
     const js_handle = _js_handle_free_list.length ? _js_handle_free_list.pop() : _next_js_handle++;
     // note _cs_owned_objects_by_js_handle is list, not Map. That's why we maintain _js_handle_free_list.
     _cs_owned_objects_by_js_handle[<number>js_handle!] = js_obj;
-    js_obj[cs_owned_js_handle_symbol] = js_handle;
+    if (Object.isExtensible(js_obj)) {
```

Comment:
this is a good thing to do, but i feel like the else case probably needs to throw. otherwise the user is silently getting weird and bad behavior

Reply:
another option would be to store the handles in a weakmap using the non-extensible object as a key.

Reply:
The consequence of else case is that we could have multiple JSHandles and multiple proxy instances. Throwing exception would prevent you from creating any proxy of it. If we have weakmap instead, we would pay the price of the lookup for all, not just non-extensible objects.

I'm OK with this trade-off and I will document it in the code. Please let me know if you disagree.